src/components/smartSearch/filters/PersonTags/index.tsx

```diff
@@ -0,0 +1,165 @@
+        if (conditionValue === MIN_MATCHING) {
+            setConfig({
+                ...filter.config,
+                condition: CONDITION_OPERATOR.ANY,
+                min_matching: minMatching,
+            });
+        }
+        else {
+            setConfig({
+                ...filter.config,
+                condition: conditionValue as CONDITION_OPERATOR,
+                min_matching: undefined,
+            });
+        }
+    };
+
+    const handleTagChange = (tags: ZetkinTag[]) => {
+        setConfig({ ...filter.config, tags: tags.map(t => t.id) });
+    };
+
+    return (
+        <form onSubmit={ e => handleAddFilter(e) }>
+            <Typography style={{ lineHeight: 'unset', marginBottom: '2rem' }} variant="h4">
+                <Msg id="misc.smartSearch.person_tags.inputString" values={{
+                    addRemoveSelect: (
+                        <StyledSelect onChange={ e => setOp(e.target.value as OPERATION) }
+                            value={ filter.op }>
+                            { Object.values(OPERATION).map(o => (
+                                <MenuItem key={ o } value={ o }>
+                                    <Msg id={ `misc.smartSearch.person_tags.addRemoveSelect.${o}` }/>
+                                </MenuItem>
+                            )) }
+                        </StyledSelect>
+                    ),
+                    condition: (
+                        <Msg
+                            id={ `misc.smartSearch.condition.edit.${selected}` }
+                            values={{
+                                conditionSelect: (
+                                    <StyledSelect
+                                        onChange={ e => handleConditionChange(e.target.value) }
+                                        value={ selected }>
+                                        { Object.values(CONDITION_OPERATOR).map(o => (
+                                            <MenuItem key={ o } value={ o }>
+                                                <Msg
+                                                    id={ `misc.smartSearch.condition.conditionSelect.${o}` }
+                                                />
+                                            </MenuItem>
+                                        )) }
+                                        <MenuItem key={ MIN_MATCHING } value={ MIN_MATCHING }>
+                                            <Msg
+                                                id="misc.smartSearch.condition.conditionSelect.min_matching"
+                                            />
+                                        </MenuItem>
+                                    </StyledSelect>
+                                ),
+                                minMatchingInput: (
+                                    <StyledNumberInput
+                                        inputProps={{ max: filter.config.tags.length, min: '1' }}
```

Comment:
Nice with the `max`!

By adding this amazing little feature, you have prompted me to find this bug. 😉 

1. Pick "at least"
2. Add two tags
3. Change min input to 2
4. Remove one of the tags

Expected results: min input decreases to 1
Actual results: min input remains at 2

Reply:
fixing this bug introduced another, so i'm going to move on for now and add it to the refactors list!